e2xgrader/server_extensions/formgrader/handlers.py

```diff
@@ -1,10 +1,106 @@
 import os
-
 from tornado import web
 from nbgrader.api import MissingEntry
+from nbgrader.apps.api import NbGraderAPI
 from nbgrader.server_extensions.formgrader.base import BaseHandler, check_xsrf, check_notebook_dir
 from nbgrader.server_extensions.formgrader.handlers import SubmissionNavigationHandler as NbgraderSubmissionNavigationHandler
-from ...exporters import GradeTaskExporter, GradeNotebookExporter, GradeAssignmentExporter
+from ...exporters import GradeTaskExporter, GradeNotebookExporter, GradeAssignmentExporter, GradeSelectedAssignmentExporter
+from e2xgrader.apps.api import E2X_Gradebook
+import json
+from multiprocessing import Process, Value
+
+
+autograde_flag = Value('b', False)
+autograde_progress = Value('d', 0.0)
+
+
+def autograder(assignment_id):
```

Comment:
Please move this function to e2xgrader/apps/api.py and rename it to autograde_all. See also https://github.com/jupyter/nbgrader/blob/master/nbgrader/apps/api.py#L1012

This way you also don't need an extra reference to the api or the gradebook and can just do ```self.gradebook```

Reply:
Moved 'autograder()' function to api.py and renamed to 'autograde_all().